perceval/backends/core/pagure.py

```diff
@@ -0,0 +1,404 @@
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = int(item['last_updated'])
+        ts = datetime.fromtimestamp(ts).timestamp()
+
+        return ts
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Pagure item.
+
+        This backend generates one type of item which is
+        'issue'.
+        """
+        category = CATEGORY_ISSUE
+
+        return category
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return PagureClient(self.namespace, self.repository, self.api_token,
+                            self.sleep_time, self.max_retries, self.max_items,
+                            self.archive, from_archive, self.ssl_verify)
+
+    def __fetch_issues(self, from_date, to_date):
+        """Fetch the issues
+        :param from_date: starting date from which issues are fetched
+        :param to_date: ending date till which issues are fetched
+
+        :returns: an issue object
+        """
+        issues_groups = self.client.issues(from_date=from_date)
+
+        for raw_issues in issues_groups:
+            issues = json.loads(raw_issues)
+            issues = issues['issues']
+            for issue in issues:
+
+                if int(issue['last_updated']) > str_to_datetime(to_date).timestamp():
+                    return
+
+                yield issue
+
+
+class PagureClient(HttpClient, RateLimitHandler):
+    """Client for retieving information from Pagure API
+
+    :param namespace: Pagure namespace
+    :param repository: Pagure repository; incase the repository is within a namespace
+    :param token: Pagure API token to access the API
+    :param sleep_time: time to sleep in case
+        of connection problems
+    :param max_retries: number of max retries to a data source
+        before raising a RetryError exception
+    :param max_items: max number of category items (e.g., issues,
+        pull requests) per query
+    :param archive: collect issues already retrieved from an archive
+    :param from_archive: it tells whether to write/read the archive
+    :param ssl_verify: enable/disable SSL verification
+    """
+    EXTRA_STATUS_FORCELIST = [403, 500, 502, 503]
+
+    def __init__(self, namespace, repository, token,
+                 sleep_time=DEFAULT_SLEEP_TIME, max_retries=MAX_RETRIES,
+                 max_items=MAX_CATEGORY_ITEMS_PER_PAGE, archive=None, from_archive=False, ssl_verify=True):
+        self.namespace = namespace
+        self.repository = repository
+        self.token = token
+        self.max_items = max_items
+
+        # URL to fetch the data from
+        base_url = PAGURE_API_URL
+
+        super().__init__(base_url, sleep_time=sleep_time, max_retries=max_retries,
+                         extra_headers=self._set_extra_headers(),
+                         extra_status_forcelist=self.EXTRA_STATUS_FORCELIST,
```

Comment:
extra_status_forcelist can be removed. We can rely on the default status (https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/client.py#L81)